libraries/alicloud_ecs_instance.rb

```diff
@@ -76,7 +76,7 @@ def fetch_instance(opts)
         action: "DescribeInstances",
         params: {
           'RegionId': opts[:region],
-          'InstanceIds': [opts[:instance_id]].to_json,
```

Comment:
This is the standard way in ruby to format a string in JSON.. Could you give me your examples so we could see what is wrong with this?

Reply:
Sent via email!

Reply:
okay, as per email they don't want this format `"[\"#{opts[:instance_id]}\"]"`, so it should be `[opts[:instance_id]]`
Could we try this?

Reply:
If you look at https://github.com/inspec/inspec-alicloud/pull/69/files, this is the issue we were trying to fix originally - that format was not working, and the .to_json was the suggested fix per the review on that PR which also didn't work

Reply:
Per the email they want the quote to be preserved inside the string without an escape character being included, whereas to_json was including the slash in the actual output string.

Reply:
okay I will investigate and get back on this issue.. because the suggested fix again will way not work ..

Reply:
Hey Jon, I just went through the internals of alicloud SDK and they are doing POST request with params as url encoded ..
now when we lets assume the params is `{:ids=>["i-bp67acfmxazb4p****", "i-bp67acfmxazb4p****", "i-bp67acfmxazb4p****"]}`

- when we submit the params like this
`[opts[:instance_id]]` it will be encoded to `"ids.1=i-bp67acfmxazb4p%2A%2A%2A%2A&ids.2=i-bp67acfmxazb4p%2A%2A%2A%2A&ids.3=i-bp67acfmxazb4p%2A%2A%2A%2A"`

- when we submit as `[opts[:instance_id]].to_json`  we get below
`ids=%5B%22i-bp67acfmxazb4p%2A%2A%2A%2A%22%2C%22i-bp67acfmxazb4p%2A%2A%2A%2A%22%2C%22i-bp67acfmxazb4p%2A%2A%2A%2A%22%5D` which when decoded returns valid json array 

- when we submit as `[\"#{opts[:instance_id]}\"]`
we get `"ids=%5B%22%5B%22i-bp67acfmxazb4p%2A%2A%2A%2A%22%2C%20%22i-bp67acfmxazb4p%2A%2A%2A%2A%22%2C%20%22i-bp67acfmxazb4p%2A%2A%2A%2A%22%5D%22%5D"` which when decoded returns nested array

also looking at the SDK logic, first option seems to be the best suitable one..

Reply:
seems this works for customer.. so approving it ..